src/fmt/pl.c

```diff
@@ -784,16 +784,20 @@ const char *pl_strstr(const struct pl *pl, const char *str)
  */
 int pl_ltrim(struct pl *pl)
 {
-	if (!pl_isset(pl))
+	if (!pl)
 		return EINVAL;
 
-	while (!re_regex(pl->p, 1, "[ \t\r\n]")) {
-		++pl->p;
-		--pl->l;
-		if (!pl->l)
-			return EINVAL;
+	if (!pl_isset(pl))
+		return 0;
+
+	size_t i = 0;
+	while (i < pl->l && isspace((unsigned char)pl->p[i])) {
+		++i;
 	}
 
+	pl->p += i;
+	pl->l -= i;
```

Comment:
optional: `pl_advance` can be used instead of these two lines ...

Reply:
Perfect, thank you for the suggestion. Done.